src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -332,4 +332,61 @@ private boolean isMatchingByLineAndColumnAndTokenType(DetailAST ast) {
                 && expandedTabColumn(ast) == columnNumber
                 && (tokenType == 0 || tokenType == ast.getType());
     }
+
+    /**
+     * Controls true escape characters. True escape characters need '&' before but it also
```

Comment:
`True escape characters need '&' before but it also`
here & should be escaped as well

Reply:
On it

Reply:
Done